lazy_dataset/core.py

```diff
@@ -31,9 +36,9 @@ def new(examples: Union[list, dict], immutable_warranty: str = 'pickle'):
         Create a dataset from a dict:
 
         >>> import lazy_dataset
-        >>> ds = lazy_dataset.new({'a': 1, 'b': 2, 'c': 3})
+        >>> ds = lazy_dataset.new({'a': 1, 'b': 2, 'c': 3}, name='MyDataset')
         >>> ds
-          DictDataset(len=3)
+          DictDataset(name=MyDataset, len=3)
```

Comment:
I would have expected the repr output here, i.e.
DictDataset(name='MyDataset', len3)

Reply:
Done